daemon/build.go

```diff
@@ -75,15 +76,24 @@ func (rl *releaseableLayer) Release() error {
 	if rl.released {
 		return nil
 	}
+	if err := rl.releaseRWLayer(); err != nil {
+		return err
+	}
+	if err := rl.releaseROLayer(); err != nil {
```

Comment:
That's a good point... and drew my attention to the code below in newReleasableLayerForImage where the roLayer is actually mounted via driver.Get... meaning we still have another place where this code mounts without unmounting.  It just happens to work because of the fact that Get and Put are no-ops on Windows read-only layers, but could have negative effects in other graphdrivers that don't differentiate between read vs read/write layers.  I'll fix this stuff and get a new PR.